springfox-swagger2/src/main/java/springfox/documentation/swagger2/web/Swagger2Controller.java

```diff
@@ -63,29 +63,30 @@
   private ServiceModelToSwagger2Mapper mapper;
 
   @Autowired
-  private JsonSerializer jsonSerializer;
+  private MultiFormatSerializer multiFormatSerializer;
 
   @ApiIgnore
   @RequestMapping(value = "${springfox.documentation.swagger.v2.path:" + DEFAULT_URL + "}",
       method = RequestMethod.GET, produces = { APPLICATION_JSON_VALUE, HAL_MEDIA_TYPE })
   public
   @ResponseBody
-  ResponseEntity<Json> getDocumentation(
+  ResponseEntity<RawOutput> getDocumentation(
+      @RequestParam(value = "format", required = false) String format,
```

Comment:
Why not make this driven by the accept header?

`@RequestHeader(value = "Accept",  defaultValue="application/json") String format,` and add the supported media types to the operations `@RequestMapping`.